Frontend/nix/watch.nix

```diff
@@ -0,0 +1,115 @@
+{ ... }:
+{
+  imports = [
+    ../node
+  ];
+
+  perSystem = { config, self', lib, pkgs, system, ... }:
+    let
+      webpack-dev-server = pkgs.writeShellApplication {
+        name = "webpack-dev-server-wrapped";
+        runtimeInputs = [ self'.packages.nodejs ];
+        text = ''
+          export NODE_PATH="${self'.packages.nodeDependencies}/node_modules"
+          PIPE="$1"
+          shift
+          tail -n1 -f "$PIPE" | node ${./watch.js} "$@"
+        '';
+      };
```

Comment:
Watch.nix is the unit of isolation in my view, see below for the more detailed comment.

The reason for a single monolithic nodeDependencies is because of webpack. The only thing that could actually be removed would be webpack-dev-server/webpack-cli. The way that webpack works means that we need to have a single node_modules folder with both the dev dependencies for running webpack and the runtime dependencies that will be bundled into the package. This is not really something that can be broken apart. Theoretically we could make two different package.json files one for each dev dependencies and runtime dependencies but dream2nix already independently compiles everything so we would not gain anything performance wise.